Fixed #31310 -- Fixed hints in checks for using intermediate model with ambiguous foreign key.

matheuscmotta opened the pull request:
Changed ForeignKey to ManyToManyField and removed outdated symmetrical=False keyword argument.

felixxm commented:
@matheuscmotta Thanks :+1: It looks that we don't have a test for `fields.E334`, can you add it in the first commit, and fix hints in the second?

matheuscmotta commented:
Done @felixxm. Added the test in first commit with the old hint structure so that tests pass at that point in timeline.

On second commit made the fixes to hints. Also changed the name of the test for field 335, so that it's consistent with the new 334 naming and provide better context on its differences: from and to.

If the renaming is undesirable or cause issues I'm unaware of, could revert it back to old name.

felixxm commented:
@matheuscmotta Thanks :+1: I pushed minor edits.
